src/pages/test-report/components/vehicle-check/vehicle-check.ts

```diff
@@ -29,34 +43,93 @@ export class VehicleCheckComponent implements OnInit {
       select(getVehicleChecks),
     );
 
-    vehicleChecks$.subscribe((vehicleChecks: VehicleChecks) => {
-      this.tellMeQuestionFault = vehicleChecks.tellMeQuestion.outcome;
-    });
+    const isSeriousMode$ = this.store$.pipe(
+      select(getTestReportState),
+      select(isSeriousMode),
+    );
+
+    const merged$ = merge(
+      vehicleChecks$.pipe(map((vehicleChecks: VehicleChecks) => {
+        this.tellMeQuestionFault = vehicleChecks.tellMeQuestion.outcome;
+        this.showMeQuestionFault = vehicleChecks.showMeQuestion.outcome;
+      })),
+      isSeriousMode$.pipe(map(toggle => this.isSeriousMode = toggle)),
+    );
 
+    this.subscription = merged$.subscribe();
   }
 
   ngOnDestroy(): void {
+    if (this.subscription) {
+      this.subscription.unsubscribe();
+    }
   }
 
   onTap = () => {
-    console.log('competency tapped');
+    this.addOrRemoveFault();
   }
 
   onPress = () => {
-    console.log('competency pressed');
+    this.addOrRemoveFault(true);
   }
 
   canButtonRipple = () => {
+    // TODO: Implement this
     return true;
   }
 
+  addOrRemoveFault = (wasPress: boolean = false): void => {
+    if (this.isRemoveFaultMode) {
+      this.removeFault();
+      return;
+    }
+
+    this.addFault(wasPress);
+  }
+
+  removeFault = (): void => {
+    // TODO: Implement this
```

Comment:
again - is this to be done in a separate PR?

Reply:
yes